app/src/main/kotlin/com/wire/android/ui/debug/DebugDataOptions.kt

```diff
@@ -17,46 +17,386 @@
  */
 package com.wire.android.ui.debug
 
+import android.content.Context
 import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.stringResource
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.BuildConfig
 import com.wire.android.R
+import com.wire.android.datastore.GlobalDataStore
+import com.wire.android.di.CurrentAccount
+import com.wire.android.migration.failure.UserMigrationStatus
 import com.wire.android.model.Clickable
+import com.wire.android.navigation.BackStackMode
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.common.RowItemTemplate
+import com.wire.android.ui.common.WireSwitch
+import com.wire.android.ui.common.button.WirePrimaryButton
+import com.wire.android.ui.common.dimensions
 import com.wire.android.ui.home.conversationslist.common.FolderHeader
 import com.wire.android.ui.home.settings.SettingsItem
+import com.wire.android.ui.theme.wireColorScheme
+import com.wire.android.ui.theme.wireTypography
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.keypackage.MLSKeyPackageCountResult
+import com.wire.kalium.logic.feature.keypackage.MLSKeyPackageCountUseCase
+import com.wire.kalium.logic.sync.incremental.RestartSlowSyncProcessForRecoveryUseCase
+import com.wire.kalium.logic.sync.periodic.UpdateApiVersionsScheduler
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+//region DebugDataOptionsVewModel
+data class DebugDataOptionsState(
+    val isEncryptedProteusStorageEnabled: Boolean = false,
+    val keyPackagesCount: Int = 0,
+    val mslClientId: String = "null",
+    val mlsErrorMessage: String = "null",
+    val isManualMigrationAllowed: Boolean = false
+)
+
+@Suppress("LongParameterList")
+@HiltViewModel
+class DebugDataOptionsVewModel
```

Comment:
```suggestion
class DebugDataOptionsViewModel
```
Typo 😄

Reply:
Also, we usually keep ViewModel and State data class in a separate class file

Reply:
That seems like a ton of overhead and switching files for something this small :)